ruby/bps-kafka/lib/bps/kafka.rb

```diff
@@ -4,34 +4,52 @@
 module BPS
   module Kafka
     # ReliablePublisher is a slow, but reliable Kafka Publisher.
-    # There is no batching at all, messages are emitted synchronously.
+    # There is no batching at all, messages are delivered one by one.
     class ReliablePublisher < BPS::Publisher::Abstract
-      DELIVER_MESSAGE_RETRIES = 3 # 1 by kafka-ruby's Kafka::Client default, increase slightly
-
       class Topic < BPS::Publisher::Topic::Abstract
-        def initialize(client, topic_name)
-          @client = client
-          @topic_name = topic_name
+        def initialize(producer, topic_name, opts = {})
+          @producer = producer
+          @opts = opts.merge(topic: topic_name)
         end
 
         def publish(msg_data)
```

Comment:
I don't see why we shouldn't do: `def publish(msg_data, **opts)` here and merge options via `          @producer.produce(msg_data, **@opts, **opts)`

Reply:
These opts would be common for ALL the publish calls (99% sure), but Ok, I'll change.